pkg/utils/resource/update/update.go

```diff
@@ -0,0 +1,72 @@
+package update
+
+import (
+	"context"
+	"reflect"
+
+	cco "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/aws-load-balancer-operator/pkg/utils"
+)
+
+// CredentialsRequest updates the given CredentialsRequest to the desired state if needed.
+// Returns true if the CredentialsRequest was updated, false otherwise.
+func CredentialsRequest(ctx context.Context, client client.Client, current *cco.CredentialsRequest, desired *cco.CredentialsRequest) (bool, error) {
+	if changed, err := IsCredentialsRequestChanged(current, desired); err != nil {
+		return false, err
+	} else if !changed {
+		return false, nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
```

Comment:
:exploding_head: You can update the name and namespace of an object after it's been created? I didn't think it was possible, I thought they were immutable.
Also, I don't see a unit test for either namespace or name change, I think that would be important to test.

Reply:
Indeed, it's unclear what was expected ho happen here if either of them would be different. I removed these assignments and added some logic to treat the name/namespace mismatch as an input error.